microcloud/cmd/microcloud/main_init.go

```diff
@@ -179,20 +179,27 @@ func lookupPeers(s *service.Handler, autoSetup bool, iface *net.Interface, subne
 		}()
 	}
 
-	var timeAfter <-chan time.Time
+	timeoutDuration := time.Minute
 	if autoSetup {
-		timeAfter = time.After(5 * time.Second)
+		timeoutDuration = 5 * time.Second
 	}
 
-	if len(expectedSystems) > 0 {
-		timeAfter = time.After(1 * time.Minute)
-	}
+	timeAfter := time.After(timeoutDuration)
 
 	expectedSystemsMap := make(map[string]bool, len(expectedSystems))
 	for _, system := range expectedSystems {
 		expectedSystemsMap[system] = true
 	}
 
+	ipv6Supported, err := mdns.CheckIPv6Status()
+	if err != nil {
+		return fmt.Errorf("Could not check IPv6 status: %v", err)
+	}
+
+	if !ipv6Supported {
+		fmt.Println("IPv6 is not supported on this system, disabling IPv6 mDNS")
```

Comment:
I'm not sure if it's absolutely necessary to include a CLI message message for this. At minimum it's adding another parameter that we have to keep track of. This might be a bit cleaner if `CheckIPv6Status()` is instead called within `LookupPeers` (or `Lookup`) and maybe set a warning or info log if IPv6 is disabled.

Reply:
fixed